bbq/compiler/compiler.go

```diff
@@ -1457,7 +1457,13 @@ func (c *Compiler[_, _]) VisitReferenceExpression(expression *ast.ReferenceExpre
 
 func (c *Compiler[_, _]) VisitForceExpression(expression *ast.ForceExpression) (_ struct{}) {
 	c.compileExpression(expression.Expression)
-	c.codeGen.Emit(opcode.InstructionUnwrap{})
+
+	// Unwrap if optional
+	valueType := c.ExtendedElaboration.ForceExpressionType(expression)
+	if _, ok := valueType.(*sema.OptionalType); ok {
+		c.codeGen.Emit(opcode.InstructionUnwrap{})
+	}
```

Comment:
Wonder if it is safe to determine this at compile time. For e.g.: there can be optional values assign to `AnyStruct` and then trying to unwrap.
```cadence
var a: Int? = 5
var b: AnyStruct = a
var c = b!
```

Reply:
You're probably right, we should match this exactly to what the interpreter is doing.

Reply:
Should be fixed by e96db5fa3213d00d0cce61c43723a929f91007a1